chia/rpc/wallet_rpc_api.py

```diff
@@ -895,6 +896,23 @@ async def get_current_derivation_index(self, request) -> Dict[str, Any]:
 
         return {"success": True, "index": index}
 
+    async def resync_wallets(self, request) -> Dict[str, Any]:
+        assert self.service.wallet_state_manager is not None
+        fingerprint = self.service.logged_in_fingerprint
+        self.service._close()
+        log.info("Stopping wallet ...")
+        # Clean tables
+        await self.service.wallet_state_manager.clean_resync_tables()
+        log.info("Cleaned tables ...")
+        peers_close_task: Optional[asyncio.Task] = await self.service._await_closed(shutting_down=False)
```

Comment:
Why delete the tables before calling await close?

Reply:
Since the connection will be closed.

Reply:
it seems safer to delete the tables *after* the wallet has shut down. what if it's trying to access them while shutting down?

Reply:
I don't understand how you can assign the return value from `_await_closed()` to a variable of type `Optional[asyncio.Task]`. `_await_closed()` is an `async` function that returns `None`. On this line, you `await` the call to it, so presumably that would return `None`.

Reply:
It's copy from _stop_wallet()

Reply:
The requirement is to reset the database without shutdown the wallet.